validator/rpc/handlers_keymanager.go

```diff
@@ -238,3 +241,169 @@ func (s *Server) DeleteRemoteKeys(w http.ResponseWriter, r *http.Request) {
 
 	http2.WriteJson(w, RemoteKeysResponse{Data: deleter.DeletePublicKeys(req.Pubkeys)})
 }
+
+// GetGasLimit returns the gas limit measured in gwei defined for the custom mev builder by public key
+func (s *Server) GetGasLimit(w http.ResponseWriter, r *http.Request) {
+	_, span := trace.StartSpan(r.Context(), "validator.keymanagerAPI.GetGasLimit")
+	defer span.End()
+
+	if s.validatorService == nil {
+		http2.HandleError(w, "Validator service not ready", http.StatusServiceUnavailable)
+		return
+	}
+
+	rawPubkey := mux.Vars(r)["pubkey"]
+	if rawPubkey == "" {
+		http2.HandleError(w, "pubkey is required in URL params", http.StatusBadRequest)
+		return
+	}
+
+	pubkey, valid := shared.ValidateHex(w, "pubkey", rawPubkey, fieldparams.BLSPubkeyLength)
+	if !valid {
+		return
+	}
+
+	resp := &GetGasLimitResponse{
+		Data: &GasLimitMetaData{
+			Pubkey: rawPubkey,
+		},
+	}
+	settings := s.validatorService.ProposerSettings()
+	if settings != nil {
+		proposerOption, found := settings.ProposeConfig[bytesutil.ToBytes48(pubkey)]
+		if found {
+			if proposerOption.BuilderConfig != nil {
+				resp.Data.GasLimit = fmt.Sprintf("%d", proposerOption.BuilderConfig.GasLimit)
+				http2.WriteJson(w, resp)
+				return
+			}
+		} else if settings.DefaultConfig != nil && settings.DefaultConfig.BuilderConfig != nil {
+			resp.Data.GasLimit = fmt.Sprintf("%d", s.validatorService.ProposerSettings().DefaultConfig.BuilderConfig.GasLimit)
+			http2.WriteJson(w, resp)
+			return
+		}
+	}
+	resp.Data.GasLimit = fmt.Sprintf("%d", params.BeaconConfig().DefaultBuilderGasLimit)
+	http2.WriteJson(w, resp)
+}
+
+// SetGasLimit updates GasLimt of the public key.
+func (s *Server) SetGasLimit(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "validator.keymanagerAPI.SetGasLimit")
+	defer span.End()
+
+	if s.validatorService == nil {
+		http2.HandleError(w, "Validator service not ready", http.StatusServiceUnavailable)
+		return
+	}
+
+	rawPubkey := mux.Vars(r)["pubkey"]
+	if rawPubkey == "" {
+		http2.HandleError(w, "pubkey is required in URL params", http.StatusBadRequest)
+		return
+	}
+
+	pubkey, valid := shared.ValidateHex(w, "pubkey", rawPubkey, fieldparams.BLSPubkeyLength)
+	if !valid {
+		return
+	}
+
+	var req SetGasLimitRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		http2.HandleError(w, "Could not decode request body: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	gasLimit, valid := shared.ValidateUint(w, "Gas Limit", req.GasLimit)
+	if !valid {
+		return
+	}
+
+	settings := s.validatorService.ProposerSettings()
+	if settings == nil {
+		http2.HandleError(w, "no proposer settings were found to update", http.StatusInternalServerError)
```

Comment:
These should all be capitalized
```suggestion
		http2.HandleError(w, "No proposer settings were found to update", http.StatusInternalServerError)
```